bindings/java/src/main/native/Makefile.am

```diff
@@ -62,7 +62,7 @@ libjucx_la_SOURCES = context.cc \
                      ucs_constants.cc \
                      worker.cc
 
-libjucx_la_CXXFLAGS = -fPIC -DPIC -Werror
+libjucx_la_CXXFLAGS = -fPIC -DPIC -Werror -std=c++98
```

Comment:
why is this needed?

Reply:
Thought to explicitly force c++98, even when compiling on  new compilers.

Reply:
https://stackoverflow.com/a/51396583/3271168
```
For example, in RHEL 6 g++ (GCC) 4.4.7 20120313 (Red Hat 4.4.7-23), the output:

         gnu++98
           GNU dialect of -std=c++98.  This is the default for C++ code.
And in Fedora 28 g++ (GCC) 8.1.1 20180502 (Red Hat 8.1.1-1), the output:

       gnu++1y
           GNU dialect of -std=c++14.  This is the default for C++ code.  The name gnu++1y 
```

Reply:
seems like a non-portable flag..

Reply:
`-std=gnu++98` supported by gcc, clang, icc.

Reply:
okay